pkg/cloudprovider/aws/packing/packing.go

```diff
@@ -63,104 +54,41 @@ func NewPacker(ec2 ec2iface.EC2API) Packer {
 // the same zone as tightly as possible. It follows the First Fit Decreasing bin
 // packing technique, reference-
 // https://en.wikipedia.org/wiki/Bin_packing_problem#First_Fit_Decreasing_(FFD)
-func (p *podPacker) Pack(ctx context.Context, constraints *cloudprovider.Constraints) ([]*Packing, error) {
-	// TODO use ctx when calling ec2 API
+func (p *packer) Pack(ctx context.Context, pods []*v1.Pod, instanceTypes []*ec2.InstanceTypeInfo) []*Packing {
 	// Sort pods in decreasing order by the amount of CPU requested, if
 	// CPU requested is equal compare memory requested.
-	sort.Sort(sort.Reverse(binpacking.ByResourcesRequested{SortablePods: constraints.Pods}))
+	sort.Sort(sort.Reverse(binpacking.ByResourcesRequested{SortablePods: pods}))
 	var packings []*Packing
 	var packing *Packing
-	var err error
-	remainingPods := constraints.Pods
+	remainingPods := pods
+	nodeCapacities := []*nodeCapacity{}
+	for _, instanceTypeInfo := range instanceTypes {
```

Comment:
Prefer the idiom

```for _, foo := range foos```
or 
```for _, instanceType := range instanceTypes```
Otherwise I need to be aware of another concept.